capstone/src/test/java/com/google/step/servlets/PageViewServletTest.java

```diff
@@ -0,0 +1,407 @@
+        request.setParameter("year", "1");
+
+        try {
+            sc.getResponse(request);
+        } catch (HttpException e) {
+            Assert.assertEquals("Error code", HttpServletResponse.SC_FORBIDDEN, e.getResponseCode());
+        }
+    }
+
+    @Test
+    public void testDoPostExistingEntity() throws SAXException, IOException {
+        // Tests the doPost when the entity that we are making the request to already exists,
+        // thus that entity count should be updated from 199 to 200.
+
+        // Calendar is needed because it is used by the putPageView method in the client.
+        Calendar calendar = Calendar.getInstance();
+        int year = calendar.get(Calendar.YEAR);
+        int week = calendar.get(Calendar.WEEK_OF_YEAR);
+
+        Entity entity = new Entity("PageViews");
+        entity.setProperty("restaurantName", "Marlows");
+        entity.setProperty("restaurantKey", "1");
+        entity.setProperty("year", year);
+        entity.setProperty("week", week);
+        entity.setProperty("count", 199);
+        datastoreService.put(entity);
+
+        ServletRunner sr = new ServletRunner();
+        sr.registerServlet("page-views", PageViewServlet.class.getName());
+        ServletUnitClient sc = sr.newClient();
+
+        // Make the request.
+        WebRequest request = new PostMethodWebRequest("http://localhost:8080/page-views");
+        request.setParameter("restaurantName", "Marlows");
+        request.setParameter("restaurantKey", "1");
+        sc.getResponse(request);
+
+        // Check if the put worked.
+        Query query = new Query("PageViews").setFilter(new Query.FilterPredicate("restaurantKey",
+                Query.FilterOperator.EQUAL, "1"));
+        PreparedQuery results = datastoreService.prepare(query);
+        Entity result = results.asSingleEntity();
+
+        Assert.assertEquals("count", "200", result.getProperty("count").toString());
+
+        // Test again to make sure it supports multiple put requests.
+        sc.getResponse(request);
+        query = new Query("PageViews").setFilter(new Query.FilterPredicate("restaurantKey",
+                Query.FilterOperator.EQUAL, "1"));
+        results = datastoreService.prepare(query);
+        result = results.asSingleEntity();
+
+        Assert.assertEquals("count", "201", result.getProperty("count").toString());
+    }
+
+    @Test
+    public void testDoPostCreateEntity() throws SAXException, IOException {
+        // Tests the doPost by having to create the entity which we are calling the doPost request for.
+
+        ServletRunner sr = new ServletRunner();
+        sr.registerServlet("page-views", PageViewServlet.class.getName());
+        ServletUnitClient sc = sr.newClient();
+
+        // Make the request.
+        WebRequest request = new PostMethodWebRequest("http://localhost:8080/page-views");
+        request.setParameter("restaurantName", "Marlows");
+        request.setParameter("restaurantKey", "1");
```

Comment:
Also update tests to not have the restaurantName param for a post.

Reply:
Yup I'm just waiting for the other PR to be accepted and rebase the changes.

Reply:
Alright done!